### DISCUSSION TITLE
Error with AMP Route?

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#259] [Q&A]
kang, ini route amp nya tidak berjalan sebagaimana mestinya, error nya dimana ya kang?

``` php
Hook::set('route', function($content, $path, $query, $hash) {
    if ($path && str_starts_with($path . '/', '/amp/')) {
        State::set('is.mobile', true);
        return Hook::fire('route', [$content, substr($path, 4), $query, $hash]);
    }
    return $content;
}, 19);

From::_('mobileURL', function(string $path) use($url) {
    // Empty URL
    if ("" === $path) {
        return $path;
    }
    // Relative URL
    if ('/amp' === $path) {
        return '/';
    }
    // Relative URL
    if (str_starts_with($path, '/amp/')) {
        return substr($path, 4);
    }
    // Full URL
    if ($url . '/amp' === $path) {
        return substr($path, 0, -4);
    }
    return strtr($path, [
        $url . '/amp/' => $url . '/'
    ]);
});

To::_('mobileURL', function(string $path) use($url) {
    // Empty URL
    if ("" === $path) {
        return '/amp';
    }
    // Relative URL
    if (str_starts_with($path, '/')) {
        return '/amp' . $path;
    }
    // Full URL
    if ($url . "" === $path) {
        return $path . '/amp';
    }
    return strtr($path, [
        $url . '/' => $url . '/amp/'
    ]);
});

```

Screenshot:
![image](https://user-images.githubusercontent.com/96925342/215379255-2b39e5ea-5ba8-4be3-94a6-48d736caf04b.png)

**Comment by taufik-nurrohman** [comment]
Coba cek di setiap berkas dengan editor kode ada format tertulis &ldquo;UTF-8 with BOM&rdquo; atau enggak.

**Reply by yuliakbard** [reply]
yang mana kang? yang ini kah

Screenshoot:
![image](https://user-images.githubusercontent.com/96925342/215384562-2500f05e-29a6-4a57-970f-d0ed28f41f57.png)

**Reply by taufik-nurrohman** [reply]
Iya tapi yang ada akhiran _BOM_.

**Reply by yuliakbard** [reply]
gak ada kang

**Reply by taufik-nurrohman** [reply]
Kalau mas pakai ekstensi _minify_ coba dinonaktifkan terlebih dahulu kemudian kasih tahu bentuk entitas HTML di kode sumber seperti apa?

**Reply by yuliakbard** [reply]
Ini ane install mecha standar aja kang dengan ekstensi user dan panel.

untuk script amp nya ane tambahin di file index.php pada layout bawaan

Screenshot:
![image](https://user-images.githubusercontent.com/96925342/216388639-331ff517-f5bf-4d9c-b198-8f5956f2ee89.png)
